Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/ColorManagement/GeneratedTransforms/Srgb_To_LinearSrgb.azsli

```diff
@@ -11,5 +11,25 @@
 real3 Srgb_To_LinearSrgb(real3 color)
 {
     // Copied from CryFx's ToAccurateLinear()
+#ifdef PRE_HLSL_2021
+    // See /o3de/Gems/Atom/Feature/Common/Assets/ShaderLib/PRE_HLSL_2021.md for details.
+    return (color.xyz < 0.04045) ? color.xyz / 12.92h : pow((color.xyz + real3(0.055, 0.055, 0.055)) / real3(1.055, 1.055, 1.055), 2.4);
+#else
     return select((color.xyz < 0.04045), color.xyz / 12.92h, pow((color.xyz + real3(0.055, 0.055, 0.055)) / real3(1.055, 1.055, 1.055), 2.4));
+#endif
+}
```

Comment:
What I meant with my previous comment is to have the same method name but different type. So you will need the following two versions with same function name. 

half3 Srgb_To_LinearSrgb(half3 color)
and
float3 Srgb_To_LinearSrgb(float3 color)
No reals needed here. And the same applies to LinearSrgb_To_Srgb

Reply:
Oh, sorry: Was thrown off by this comment in the code:
```
/* AZSL doesn't support function overload for the moment.
half3 LinearToSRGB(half3 col)
```
Fixed it now.